sdk/iOS/OoyalaSkinSDK/OOSkinViewController.m

```diff
@@ -243,18 +242,17 @@ - (void)dealloc {
   [self.player destroy];
 }
 
-@end
-
-@implementation OOSkinViewController(Internal)
-
-- (void)toggleFullscreen {
+- (void)setFullscreen:(BOOL)fullscreen {
+  if (fullscreen == _fullscreen) return;
```

Comment:
good point. In iOS the rendering and UI state of the video player is managed by ourselves, outside of the player logic. Apple only gives a way to render video, but they don't have a concept of fullscreen, they just give you a view and you do whatever you want with it. That's why we need to maintain a state ourselves.